sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCOptions.scala

```diff
@@ -89,6 +89,10 @@ class JDBCOptions(
   // the number of partitions
   val numPartitions = parameters.get(JDBC_NUM_PARTITIONS).map(_.toInt)
 
+  // the number of seconds the driver will wait for a Statement object to execute to the given
+  // number of seconds. Zero means there is no limit.
+  val queryTimeout = parameters.getOrElse(JDBC_QUERY_TIMEOUT, "0").toInt
```

Comment:
You suggested here?
https://github.com/apache/spark/blob/6782359a04356e4cde32940861bf2410ef37f445/sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala#L257

I feel it's enough to update a doc for parameter.

Reply:
> You suggested here?

Yes, please, describe the `queryTimeout` parameter among `user`, `password` and `fetchsize` something like that:
```
"fetchsize" can be used to control the number of rows per fetch, and "queryTimeout" to restrict query execution time.
```

> I feel it's enough to update a doc for parameter.

It would be useful to have it [there](https://spark.apache.org/docs/2.3.0/api/scala/index.html#org.apache.spark.sql.DataFrameReader@jdbc(url:String,table:String,columnName:String,lowerBound:Long,upperBound:Long,numPartitions:Int,connectionProperties:java.util.Properties):org.apache.spark.sql.DataFrame) too